r2-netty/src/main/java/com/linkedin/r2/transport/http/client/common/ChannelPoolLifecycle.java

```diff
@@ -118,19 +118,24 @@ public void create(final Callback<Channel> channelCallback)
         @Override
         public void userEventTriggered(ChannelHandlerContext ctx, Object evt)
         {
-          if(evt == SslHandshakeCompletionEvent.SUCCESS){
+          if(evt == SslHandshakeCompletionEvent.SUCCESS)
+          {
             channelCallback.onSuccess(c);
             c.pipeline().remove(CHANNELPOOL_SSL_CALLBACK_HANDLER);
           }
+          else if (evt instanceof SslHandshakeCompletionEvent)
+          {
+            Throwable sslException = ((SslHandshakeCompletionEvent) evt).cause();
+            onError(channelCallback, sslException);
```

Comment:
Do we have guarantee that cause() is not null?

Reply:
Yes if it is not instance of SslHandshakeCompletionEvent.SUCCESS - it is guaranteed to have cause() in it.

https://github.com/netty/netty/blob/00afb19d7a37de21b35ce4f6cb3fa7f74809f2ab/handler/src/main/java/io/netty/handler/ssl/SslHandshakeCompletionEvent.java

But yea - it make sense to not depend on this implementation detail and I will add a null check.

Reply:
Let's add a comment here then

Reply:
Shouldn't this also be a retriable exception?

Reply:
Thank you. good point. I will address this in the next release as I already merged the pull request and Identity do not leverage retry exception.

Reply:
@FreCap  - please let me know if you have any other concerns !